### ISSUE TITLE
Js-of-ocaml code fails with "Base_am_testing not implemented"

### ISSUE THREAD

**Issue by smondet** [janestreet/ppx_inline_test#7]
![image](https://cloud.githubusercontent.com/assets/617111/24223494/c2681f8c-0f2c-11e7-8a0b-8233e27bdaf7.png)

It seems that there is: a `runtime.js` there: https://github.com/janestreet/ppx_inline_test/blob/44c2aac3a7657875ee957cd6877522f474acbca1/runner/lib/runtime.js

I don't use `ppx_inline_test` nor `Base` but I guess some dependency (most likely `ppx_sexp_conv`) is tracking one of them into my JSOO output?

How do I get that `runtime.js`?

I guess I need to set exact opam constraints for older released versions of my Web UI, when did this "global module inititialization" code appear?

**Comment by smondet** [comment]
Confirming that the upgrade to `v0.9.0` causes this, and `js_of_ocaml` reports:

```
There are some missing primitives
Dummy implementations (raising 'Failure' exception) will be used if they are not available at runtime.
You can prevent the generation of dummy implementations with the commandline option '--disable genprim'
Missing primitives:
  Base_am_testing
```

**Comment by hhugo** [comment]
this is coming from base (https://github.com/janestreet/base/blob/711aa92e22ba8904e9a247666c7caabcdd711682/src/import0.ml#L246).

You should be able to get the right `runtime.js` with 
`js_of_ocaml +base/runtime.js main.byte`

**Comment by smondet** [comment]
@hhugo So I tracked down why I am using base; and it's because now the dependencies have been "inverted" Sexplib now depends on Base http://opam.ocaml.org/packages/sexplib/sexplib.v0.9.0/. But why?

**Comment by diml** [comment]
@smondet we made this decision for two reasons:

1. so that Base can use S-expression and still have no dependency
2. so that the code of Sexplib can benefit from using Base

(2) was one of the reason we started Base. There is a lot of code in the libraries that come before Core_kernel: sexplib, bin_prot, typerep, ... And often when we improved a standard thing in Core_kernel, the code before it didn't benefited from the improvement. Having Base as the root package allows to desgin all of the codebase using the same base.
